src/sinks/pulsar.rs

```diff
@@ -51,17 +49,13 @@ pub enum Encoding {
 
 struct PulsarSink {
     encoding: EncodingConfig<Encoding>,
-    producer: Arc<Mutex<Producer<TokioExecutor>>>,
-    in_flight: FuturesUnordered<MetadataFuture<SendFuture, usize>>,
-    // ack
-    seq_head: usize,
-    seq_tail: usize,
-    pending_acks: HashSet<usize>,
+    producer: Option<Producer<TokioExecutor>>,
+    in_flight_prepare:
+        Option<BoxFuture<'static, (Producer<TokioExecutor>, Result<SendFuture, PulsarError>)>>,
```

Comment:
Will replace to `enum` tomorrow and write few explanation comments for a review. Today is too late :(